### PR TITLE
test(gruntfile,test/): Unit testing coverage for registration and bump-only

### PR DESCRIPTION
**Pull request by bitwit**
Introduce better unit testing coverage and a foundation for better issue testing
and pull request approval with test proofing

Remove nodeunit and replace with mocha in order to better group
and describe tests in plain english

### CODE REVIEW

**Comment by eddiemonge** [comment]
would it be possible to extend this and mock out git so there can be more coverage?

**Comment by bitwit** [comment]
This would require mocking out 'child_process.exec', I suppose, more than git really if all we are trying to do is test the message that went out. Possibly like this SO answer mentions with sinon.

http://stackoverflow.com/a/16308099

then test against certain grunt options, put expectations on the cli commands run, and mock responses until certain string conditions

If that makes sense, I could try it in the next few weeks.

**Comment by eddiemonge** [comment]
Yes please!

**Comment by swenzel** [comment]
Maybe you could also have a look at this:
https://www.npmjs.com/package/mock-spawn
Im sure despite the name you can also mock child_process.exec with it!

**Comment by mojoaxel** [comment]
It sad, but I'm closing this because of inactivity.
If anybody is willing to improve testing (maybe based on this pr) please feel free to open a new issue/pull request.
